src/lib/transition_frontier/full_frontier/full_frontier.ml

```diff
@@ -415,6 +418,17 @@ let move_root t ~new_root_hash ~new_root_protocol_states ~garbage
     (* we need to perform steps 4-7 iff there was a proof emitted in the scan
      * state we are transitioning to *)
     if Breadcrumb.just_emitted_a_proof new_root_node.breadcrumb then (
+      let location =
+        Persistent_root.Locations.potential_snarked_ledger
+          t.persistent_root_instance.factory.directory
+      in
+      let ledger =
+        Ledger.Db.create_checkpoint t.persistent_root_instance.snarked_ledger
```

Comment:
Is it possible to generate a checkpoint without opening a new db connection to that checkpoint?